tests/unit/package_managers/test_generic.py

```diff
@@ -0,0 +1,43 @@
+from unittest import mock
+
+import pytest
+
+from cachi2.core.errors import PackageRejected
+from cachi2.core.models.input import GenericPackageInput
+from cachi2.core.package_managers.generic import (
+    DEFAULT_LOCKFILE_NAME,
+    _resolve_generic_lockfile,
+    fetch_generic_source,
+)
+from cachi2.core.rooted_path import RootedPath
+
+
+@mock.patch("cachi2.core.package_managers.rpm.main.RequestOutput.from_obj_list")
+@mock.patch("cachi2.core.package_managers.generic._resolve_generic_lockfile")
+def test_fetch_generic_source(
```

Comment:
Ok but something like we have for example for [yarn classic](https://github.com/containerbuildsystem/cachi2/blob/main/tests/unit/package_managers/yarn_classic/test_main.py#L56) is more extendable in the future when the `resolve_generic_lockfile` will return real `Component` objects.

Reply:
I was contemplating if I should parametrize it, like other tests are, decided to go simple. Fixed now.